components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/applications/parser/DefaultApplicationParser.java

```diff
@@ -438,13 +434,13 @@ private Application buildCompositeAppStructure(ApplicationContext applicationCon
 		    arrImportMetadata = importMetadataKeys.toArray(arrImportMetadata);
 
             // Find tenant range of cluster
-            String tenantRange = AutoscalerUtil.findTenantRange(tenantId, cartridgeInfo.getTenantPartitions());
+            String tenantRange = AutoscalerUtil.findTenantRange(tenantId, cartridge.getTenantPartitions());
 		    Boolean isLB=false;
-			if(cartridgeInfo.getCategory().equals("lb")){
+			if(cartridge.getCategory().equals("lb")){
```

Comment:
As a best practice, compare a value assigned to a field against a known constant whenever possible. This would help avoid redundant null checks, etc that need to be performed prior to calling equals method upon the variable values being compared.